### ISSUE TITLE
setuptools > 19.4 is breaking holland (and it's plugins) installs

### ISSUE THREAD

**Issue by mancdaz** [rcbops/rpc-openstack#777]
This is the issue when attempting to do something like this with setuptools >= 19.4:
```
pip install plugins/holland.backup.random/
```

```
Processing ./plugins/holland.backup.random
    Complete output from command python setup.py egg_info:
    Traceback (most recent call last):
      File "<string>", line 3, in <module>
      File "/usr/local/lib/python2.7/dist-packages/setuptools/__init__.py", line 11, in <module>
        from setuptools.extern.six.moves import filterfalse, map
      File "/usr/local/lib/python2.7/dist-packages/setuptools/extern/__init__.py", line 1, in <module>
        from pkg_resources.extern import VendorImporter
      File "/usr/local/lib/python2.7/dist-packages/pkg_resources/__init__.py", line 3130, in <module>
        @_call_aside
      File "/usr/local/lib/python2.7/dist-packages/pkg_resources/__init__.py", line 3116, in _call_aside
        f(*args, **kwargs)
      File "/usr/local/lib/python2.7/dist-packages/pkg_resources/__init__.py", line 3155, in _initialize_master_working_set
        add_activation_listener(lambda dist: dist.activate())
      File "/usr/local/lib/python2.7/dist-packages/pkg_resources/__init__.py", line 956, in subscribe
        callback(dist)
      File "/usr/local/lib/python2.7/dist-packages/pkg_resources/__init__.py", line 3155, in <lambda>
        add_activation_listener(lambda dist: dist.activate())
      File "/usr/local/lib/python2.7/dist-packages/pkg_resources/__init__.py", line 2655, in activate
        declare_namespace(pkg)
      File "/usr/local/lib/python2.7/dist-packages/pkg_resources/__init__.py", line 2225, in declare_namespace
        _handle_ns(packageName, path_item)
      File "/usr/local/lib/python2.7/dist-packages/pkg_resources/__init__.py", line 2194, in _handle_ns
        path.sort(key=sort_key)
      File "/usr/local/lib/python2.7/dist-packages/pkg_resources/__init__.py", line 2192, in sort_key
        return sys_path.index(_normalize_cached(os.sep.join(parts)))
    ValueError: '/tmp/pip-l8H5uM-build' is not in list
```

@hughsaunders and I traced it back to this commit in setuptools which was released with 19.4:

https://bitbucket.org/pypa/setuptools/commits/1d2431a479f1737b95a96e62446a7b726d1d603e

Confirmed by installing setuptools 19.3, and running the pip install again:

```
root@kilo_galera_container-eeca50f3:/opt/holland_v1.0.10# pip install setuptools==19.3
Collecting setuptools==19.3
/usr/local/lib/python2.7/dist-packages/pip/_vendor/requests/packages/urllib3/util/ssl_.py:90: InsecurePlatformWarning: A true SSLContext object is not available. This prevents urllib3 from configuring SSL appropriately and may cause certain SSL connections to fail. For more information, see https://urllib3.readthedocs.org/en/latest/security.html#insecureplatformwarning.
  InsecurePlatformWarning
  Using cached setuptools-19.3-py2.py3-none-any.whl
Installing collected packages: setuptools
  Found existing installation: setuptools 19.4
    Uninstalling setuptools-19.4:
      Successfully uninstalled setuptools-19.4
Successfully installed setuptools-19.3
You are using pip version 7.1.2, however version 8.0.2 is available.
You should consider upgrading via the 'pip install --upgrade pip' command.
root@kilo_galera_container-eeca50f3:/opt/holland_v1.0.10# pip install plugins/holland.backup.random/
Processing ./plugins/holland.backup.random
Installing collected packages: holland.backup.random
  Running setup.py install for holland.backup.random
Successfully installed holland.backup.random-1.0.10
You are using pip version 7.1.2, however version 8.0.2 is available.
You should consider upgrading via the 'pip install --upgrade pip' command.
```

options we thought of to fix:

pin setuptools to 19.3
instead of 'pip install [directoryname]' for holland, build wheels
instead of 'pip install [directoryname] for holland, do local ```python setup.py install``` which also works

further investigation needed to try and determine if this is an actual issue with setuptools 19.4, or the way the holland package is put together. I created the following bug against setuptools in the meantime:

https://bitbucket.org/pypa/setuptools/issues/491/setuptools-194-breaks-pip-install-builds

-------------
# Committed
- [x] Master
- [x] Kilo

# Released
- [x] Kilo

**Comment by mancdaz** [comment]
or actually we can just build wheels for holland and plugins and drop them in the repo, rather than pip installing locally.

**Comment by d34dh0r53** [comment]
I think I have a fix for this, PR forthcoming

**Comment by prometheanfire** [comment]
This will be fixed in master in #750

**Comment by tomc603** [comment]
This was fixed in #796.

**Comment by prometheanfire** [comment]
setuptools has a fix in 19.6.2 so maybe we don't need the pin anymore?  will make a pr to test

**Comment by prometheanfire** [comment]
done, closing
